### ISSUE TITLE
CLI: stack name is not checked before delete confirmation

### ISSUE THREAD

**Issue by mrgum** [aws/aws-cdk#32545]
### Describe the bug

if you cdk destroy a stack that does not exist (say you are accidentally in the directory) cdk says

Are you sure you want to delete:  (y/n)?
and does nothing when you say y

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

cdk to respond
No stacks match the name(s) whatever/you/asked/for
like cdk synth does

### Current Behavior

Are you sure you want to delete:  (y/n)?

### Reproduction Steps

cdk destroy made/of/a/nonexistent/stack

### CDK CLI Version

2.173.1 (build 4eac959)

### Node.js Version

v22.11.0

### OS

Mac

### Language

Python

### Language Version

3.13

### Other information

I don't think the language is relevant

**Comment by khushail** [comment]
Hi @mrgum , thanks for reporting this.

Yes, I can confirm that while deleting the deployed stack with command `cdk destroy`, if you mention some other stack (which does not exist) being in the directory where you just deployed stack, there should be some confirmation check or warning message with the name of the stack .

Current situation -

the below mentioned stack does not exist yet the `cdk destroy` command simply exits and does not print anything
```
khushail@a07817b31bf3 testStack % cdk destroy abteststack
Are you sure you want to delete:  (y/n)? y
```

while deleting the one which exists -
```
khushail@a07817b31bf3 testStack % cdk destroy TestStackStack
Are you sure you want to delete: TestStackStack (y/n)? y
TestStackStack: destroying... [1/1]

 ✅  TestStackStack: destroyed
```

Looking at the code for `destroy` , looks like there is no check for stackname

https://github.com/aws/aws-cdk/blob/fb030d6b2a6ebd1a673b13c830389c4fc65c4dda/packages/aws-cdk/lib/cdk-toolkit.ts#L798C1-L828C8

IMO, it might include a region wide search in case , `cdk destroy nonExistentStack` will check if the mentioned stack name exists in the region, which might be time taking. 

This would be good to have feature. Marking it as P3 to keep it open for community contribution. Requesting community upvotes to get the proper traction.

**Comment by khushail** [comment]
@nmussy , I agree with you , deleting stacks outside the scope might pose a bigger risk, rather than checking stacks created in current directory/app sounds better.

**Comment by go-to-k** [comment]
I submitted a PR for an issue like this before, but there was a regression and it was reverted. I'm posting it here in case it might be of some help.

https://github.com/aws/aws-cdk/issues/27179

https://github.com/aws/aws-cdk/pull/27921

**Comment by go-to-k** [comment]
Apparently, the revert of the above PR was due to a cause unrelated to that PR. That has been corrected in [this PR](https://github.com/aws/aws-cdk/pull/31107) (see: https://github.com/aws/aws-cdk/blob/v2.173.1/packages/@aws-cdk-testing/cli-integ/tests/cli-integ-tests/cli.integtest.ts#L286-L291).
Therefore, I will submit the same PR again. This is a safe way to warn if the specified stacks don't belong to the app without searching the out-of-scope stack.

**Comment by aemada-aws** [comment]
Closing as duplicate of https://github.com/aws/aws-cdk/issues/22240
